### DISCUSSION TITLE
Upgrade to 4.10.0-0.okd-2022-07-09-073606 - initial cluster version cleanup container rm issues

### DISCUSSION THREAD

**Discussion by stanbog** [okd-project/okd#1600] [General]
**Describe the bug**

After trying to upgrade the cluster from version `4.10.0-0.okd-2022-06-10-131327` to `4.10.0-0.okd-2022-07-09-073606` , the job `version-4.10.0-0.okd-2022-07-09-073606-xxxxx `under namespace `openshift-cluster-version` failed with the following error:
`rm: invalid option -- 's'
Try 'rm ./-sts-xjA5wJZtz1bRCXw5w' to remove the file '-sts-xjA5wJZtz1bRCXw5w'.
Try 'rm --help' for more information.`

The error was generated by the `cleanup` container with the command:
`sh -c rm -fR *`
Container image: `registry.ci.openshift.org/origin/release@sha256:5a748561c829c23c7413b0fe0ec8f97b4eabf5cfd1934b9e93a8a5df3f59738d`

Apparently one of the folders on one of the master nodes that was trying to be deleted by the `cleanup` container was randomly named like:
`/sysroot/ostree/deploy/fedora-coreos/deploy/6c979558a5d3679cee96bc5d5ead6c6751ff6bd8d2cc8b261d5b6393a31eb054.0/etc/cvo/updatepayloads/-sts-xjA5wJZtz1bRCXw5w`

The `rm` command saw the `-` at the beginning of the folder name as a parameter and failed.
This can probably be easy fixable by re-building the `cleanup` container image with the command like:
`sh -c rm -fR -- *`

We managed to fix this by deleting the folder manually from the node.
Afterwards, the job completed and the upgrade continued.

**Version**

`4.10.0-0.okd-2022-06-10-131327`

**How reproducible**

`100%`

**Comment by vrutkovs** [comment]
That looks like a problem with CVO. Could you report it to https://issues.redhat.com/browse/OCPBUGS, component "Cluster Version Operator"
